extensions/wikia/Recirculation/js/helpers/sponsoredContent.js

```diff
@@ -38,9 +38,19 @@ define('ext.wikia.recirculation.helpers.sponsoredContent', [
 	}
 
 	function getApplicableContent(sponsoredContent) {
-		return sponsoredContent.filter(function (el) {
-			return !el.geos.length || el.geos.indexOf(userGeo) !== -1;
+		var applicableContent = sponsoredContent.filter(function (el) {
+			return (!el.geos.length || el.geos.indexOf(userGeo) !== -1) && (!el.wikiIds.length || el.wikiIds.indexOf(w.wgCityId) !== -1);
```

Comment:
considering the repetition, I'd write it like
```
var filterBy = function(arr, searchedItem) {
    return arr.filter(function(el) {
        return (!arr.length || arr.indexOf(searchedItem) !== -1)
    })
};

var contentByGeo = filterBy(sponsoredContent, userGeo);
var contentByWikiId = filterBy(sponsoredContent, w.wgCityId);
var contentByWikiIdAndGeo = filterBy(contentByGeo, w.wgCityId);

if (contentByWikiIdAndGeo.length) {
    return contentByWikiIdAndGeo;
}

if (contentByWikiId.length) {
    return contentByWikiId;
}

return contentByGeo;
```
bc currently 
- `applicableContent` is imo always targeted for specific wiki, so `siteSpecificContent` == `applicableContent` 
- `if (siteSpecificContent) {` will be `true` even for `[]`
- we don't support "wikiId without any geo" scenario

Reply:
what about the case when there is no content with given geo nor with given wiki ID and we should fallback to items that do not have targeting specified at all?

Reply:
it seems that the existing logic won't return a sponsored item if none match the current geo.

Reply:
i think we tested that and it worked, but anyways, we should always fall back to item that is not targeted at all if none better match is found

Reply:
@rybmat 
then we'll need to change/add
```
if (contentByGeo.length) {
    return contentByGeo;
}

return sponsoredContent;
```
to the code I've added above